crates/frontend/src/components/condition_pills/types.rs

```diff
@@ -1,6 +1,122 @@
+use std::fmt::Display;
+
+use serde_json::{Map, Value};
+
+#[derive(Debug, Clone)]
+pub enum ConditionOperator {
+    Is,
+    In,
+    Has,
+    Between,
+    Other(String),
+}
+
+impl Display for ConditionOperator {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        match self {
+            Self::Has => f.write_str("has"),
+            Self::Is => f.write_str("is"),
+            Self::In => f.write_str("in"),
+            Self::Between => f.write_str("between"),
+            Self::Other(o) => f.write_str(o),
+        }
+    }
+}
+
+impl From<(String, &Vec<Value>)> for ConditionOperator {
+    fn from(value: (String, &Vec<Value>)) -> Self {
+        let (operator, operands) = value;
+        let operand_0 = operands.get(0);
+        let operand_1 = operands.get(1);
+        let operand_2 = operands.get(2);
+        match (operator.as_str(), operand_0, operand_1, operand_2) {
+            // assuming there will be only two operands, one with the dimension name and other with the value
+            ("==", _, _, None) => ConditionOperator::Is,
+            ("<=", Some(_), Some(Value::Object(a)), Some(_)) if a.contains_key("var") => {
+                ConditionOperator::Between
+            }
+            // assuming there will be only two operands, one with the dimension name and other with the value
+            ("in", Some(Value::Object(a)), Some(_), None) if a.contains_key("var") => {
+                ConditionOperator::In
+            }
+            // assuming there will be only two operands, one with the dimension name and other with the value
+            ("in", Some(_), Some(Value::Object(a)), None) if a.contains_key("var") => {
+                ConditionOperator::Has
+            }
+            _ => ConditionOperator::Other(operator),
+        }
+    }
+}
+
 #[derive(Clone)]
 pub struct Condition {
     pub left_operand: String,
-    pub operator: String,
+    pub operator: ConditionOperator,
     pub right_operand: String,
 }
+
+impl TryFrom<&Map<String, Value>> for Condition {
+    type Error = &'static str;
+    fn try_from(source: &Map<String, Value>) -> Result<Self, Self::Error> {
+        if let Some(operator) = source.keys().next() {
+            let emty_vec = vec![];
+            let operands = source[operator].as_array().unwrap_or(&emty_vec);
+
+            let operator = ConditionOperator::from((operator.to_string(), operands));
+
+            let dimension_name = operands
+                .iter()
+                .find_map(|item| match item.as_object() {
+                    Some(o) if o.contains_key("var") => {
```

Comment:
can we have better name instead of "o"

Reply:
hey, `o` is just a throw away variable not even being used anywhere instead of that `Some()` arm, if its affecting readability I will change, if not lets go with this.
In my opinion I don't really think we need a full word for variable name for these kind of scenarios.

Reply:
you are using o in` o.contains_key` and in `o["var"]`, when you are operating on any variable it would be better if it is not just o,a,b,c,d single alphabet, 
atleast `val` would work